fl4health/clients/basic_client.py

```diff
@@ -261,12 +262,13 @@ def fit(self, parameters: NDArrays, config: Config) -> Tuple[NDArrays, int, Dict
             config (NDArrays): The config from the server.
 
         Returns:
-            Tuple[NDArrays, int, Dict[str, Scalar]]: The parameters following the local training along with the
+            Tuple[NDArrays, int, dict[str, Scalar]]: The parameters following the local training along with the
             number of samples in the local training dataset and the computed metrics throughout the fit.
 
         Raises:
             ValueError: If local_steps or local_epochs is not specified in config.
         """
+        round_start_time = datetime.datetime.now()
```

Comment:
Not exactly sure how this would look. The reports manager is called frequently every epoch, every round and every step, so it would be tough for it to track where it is in the training loop. But maybe I am missing something here

Reply:
Yeah I see what you mean. I guess I was just trying to pull this responsibility out of basic client but perhaps its not possible.